packages/react-app/src/modules/Counter/CounterDetail/CounterDetail.container.js

```diff
@@ -1,15 +1,15 @@
 import { connect } from 'react-redux'
-import { prop, path } from 'ramda'
+import { path } from 'ramda'
 
 import Component from './CounterDetail.component'
-import { incrementCounter, decrementCounter } from '../counters.reducer'
+import { incrementCounter, decrementCounter, navigateToOverview } from '../counters.reducer'
 
-const mapStateToProps = (state) => {
-  const selectedCounter = prop('selectedCounter', state)
+const mapStateToProps = (state, ownProps) => {
+  const selectedCounter = path(['match', 'params', 'selectedCounter'], ownProps)
```

Comment:
this should work:
```
const selectedCounter = prop('selectedCounter', state)
```
ie. if we refresh the page there should be middleware that calls selectCounter on some init or route action
```